src/main/java/xbot/common/injection/components/BaseComponent.java

```diff
@@ -166,14 +166,14 @@ public abstract class BaseComponent {
 
     public abstract XCANVictorSPXFactory canVictorSpxFactory();
 
-    public abstract XCANSparkMaxFactory canSparkMaxFactory();
-
     public abstract XLidarLiteFactory lidarLiteFactory();
 
     public abstract XSpeedControllerFactory speedControllerFactory();
 
     public abstract XZeromqListenerFactory zeromqListenerFactory();
 
+    public abstract XCANMotorController.XCANMotorControllerFactory motorControllerFactory();
```

Comment:
⭐ to align with the type, maybe this should be called `canMotorControllerFactory();`?